Assembler.UnitTests/FrameHandlers/FinalFrameHandlerTests.cs

```diff
@@ -78,6 +85,9 @@ public void Handle_IdentifierInCache_MessageBeingRemovedEnrichedAndReleased()
             _dateTimeProviderMock.Verify(provider => provider.Now, Times.Once);
             Assert.AreEqual(DateTime.MinValue, message.LastFrameReceived);
 
+            Assert.AreEqual(frame.Object.Guid, message.BasedOnFramesGuids.Last());
+            Assert.AreEqual(basedOnCount + 1, message.BasedOnFramesGuids.Count);
```

Comment:
You have to verify that the 2 guids that were before handling still there, and not only by count + 1. Create `expectedBasedOnFramesGuids` with `firstFrameGuid`, `secondFrameGuid` and `frame.Guid`, and use collectionAssert.Equals. That will be the prettiest way

Reply:
f